src/app/clusters/webrtc-transport-requestor-server/webrtc-transport-requestor-cluster.cpp

```diff
@@ -0,0 +1,417 @@
+            for (auto & session : mCurrentSessions)
+            {
+                ReturnErrorOnFailure(listEncoder.Encode(session));
+            }
+            return CHIP_NO_ERROR;
+        });
+    case ClusterRevision::Id:
+        return encoder.Encode(WebRTCTransportRequestor::kRevision);
+    default:
+        return Protocols::InteractionModel::Status::UnsupportedAttribute;
+    }
+}
+
+std::optional<DataModel::ActionReturnStatus> WebRTCTransportRequestorServer::InvokeCommand(const DataModel::InvokeRequest & request,
+                                                                                           TLV::TLVReader & input_arguments,
+                                                                                           CommandHandler * handler)
+{
+    ChipLogDetail(Zcl, "WebRTCTransportRequestorServer: InvokeCommand called with CommandId=0x%08" PRIx32, request.path.mCommandId);
+
+    switch (request.path.mCommandId)
+    {
+    case Commands::Offer::Id: {
+        Commands::Offer::DecodableType req;
+        if (DataModel::Decode(input_arguments, req) != CHIP_NO_ERROR)
+        {
+            return Protocols::InteractionModel::Status::InvalidCommand;
+        }
+        return HandleOffer(req, *handler);
+    }
+    case Commands::Answer::Id: {
+        Commands::Answer::DecodableType req;
+        if (DataModel::Decode(input_arguments, req) != CHIP_NO_ERROR)
+        {
+            return Protocols::InteractionModel::Status::InvalidCommand;
+        }
+        return HandleAnswer(req, *handler);
+    }
+    case Commands::ICECandidates::Id: {
+        Commands::ICECandidates::DecodableType req;
+        if (DataModel::Decode(input_arguments, req) != CHIP_NO_ERROR)
+        {
+            return Protocols::InteractionModel::Status::InvalidCommand;
+        }
+        return HandleICECandidates(req, *handler);
+    }
+    case Commands::End::Id: {
+        Commands::End::DecodableType req;
+        if (DataModel::Decode(input_arguments, req) != CHIP_NO_ERROR)
+        {
+            return Protocols::InteractionModel::Status::InvalidCommand;
+        }
+        return HandleEnd(req, *handler);
+    }
+    default:
+        return Protocols::InteractionModel::Status::UnsupportedCommand;
+    }
+}
+
+CHIP_ERROR WebRTCTransportRequestorServer::AcceptedCommands(const ConcreteClusterPath & path,
+                                                            ReadOnlyBufferBuilder<DataModel::AcceptedCommandEntry> & builder)
+{
+    return builder.ReferenceExisting(kAcceptedCommands);
+}
+
+uint16_t WebRTCTransportRequestorServer::GenerateSessionId()
+{
+    static uint16_t lastSessionId = 0;
+    uint16_t candidateId          = 0;
+
+    while (true)
+    {
+        candidateId = lastSessionId++;
+
+        // Handle wrap-around per spec
+        if (lastSessionId > kMaxSessionId)
+        {
+            lastSessionId = 0;
+        }
+
+        if (FindSession(candidateId) == nullptr)
+        {
+            break;
```

Comment:
return here instead of breaking for the purpose of returning? Makes the code flow clearer.